### DISCUSSION TITLE
Interrupt Watchdog Timer

### DISCUSSION THREAD

**Discussion by OOHehir** [LIFsCode/ELOC-3.0#102] [General]
@LIFsCode @EDsteve 
In the course of some investigation into the Interrupt Watchdog Timer (after seeing a number of crashes due to it's timeout) I found [documentation](https://docs.espressif.com/projects/esp-idf/en/stable/esp32s2/api-reference/system/wdts.html#configuration) stating:

- Note that the default timeout is higher if PSRAM support is enabled, as a critical section or interrupt routine that accesses a large amount of PSRAM takes longer to complete in some circumstances.
- The timeout should always at least twice longer than the period between FreeRTOS ticks (see [CONFIG_FREERTOS_HZ](https://docs.espressif.com/projects/esp-idf/en/stable/esp32s2/api-reference/kconfig.html#config-freertos-hz)).

Currently we have

```
CONFIG_ESP_INT_WDT_TIMEOUT_MS=300
CONFIG_FREERTOS_HZ=1000
```
& we're (depending on configuration) using PSRAM.  My question, should we increase the value of CONFIG_ESP_INT_WDT_TIMEOUT_MS above the default of 300ms?

**Comment by LIFsCode** [comment]
@EDsteve @OOHehir 

Sorry for being off for such a long time :(  I'm quite busy at the moment 

Hm yes seems strange in the depressing documentation they only state it should be higher than the default when using PSRAM but don't mention a number.

Let's increase it to 500-600 Ms this is still fast enough to capture any critical errors in the code and should help getting the system more stable

**Comment by OOHehir** [comment]
@LIFsCode No problem, I don't think I created any major disasters while you were away!

I was looking for a way of servicing/ monitoring this IWDT but no success, there doesn't seem to be an API available for such purposes. 

I try experimenting with some values.
